src/emmetHelper.ts

```diff
@@ -949,37 +949,56 @@ function getFormatters(syntax: string, preferences: any) {
 
 /**
  * Updates customizations from snippets.json and syntaxProfiles.json files in the directory configured in emmet.extensionsPath setting
+ * @param emmetExtensionsPathSetting setting passed from emmet.extensionsPath, support multiple path setting
+ * https://github.com/microsoft/vscode/issues/116741
  */
-export async function updateExtensionsPath(emmetExtensionsPath: string | undefined | null, fs: FileService, workspaceFolderPath?: URI, homeDir?: URI): Promise<void> {
-	if (emmetExtensionsPath) {
-		emmetExtensionsPath = emmetExtensionsPath.trim();
-	}
-	if (!emmetExtensionsPath) {
-		resetSettingsFromFile();
-		return Promise.resolve();
+export async function updateExtensionsPath(emmetExtensionsPathSetting: string | string[] | undefined | null, fs: FileService, workspaceFolderPath?: URI, homeDir?: URI): Promise<void> {
+	let emmetExtensionsArray: string[];
+
+	if (Array.isArray(emmetExtensionsPathSetting)){
+		emmetExtensionsArray = emmetExtensionsPathSetting;
+	} else {
+		emmetExtensionsArray = [emmetExtensionsPathSetting];
 	}
 
 	let emmetExtensionsPathUri: URI | undefined;
-	if (emmetExtensionsPath[0] === '~') {
-		if (homeDir) {
-			emmetExtensionsPathUri = joinPath(homeDir, emmetExtensionsPath.substr(1));
+	let findValidPath = false;
+	for (let emmetExtensionsPath of emmetExtensionsArray){
+		if (emmetExtensionsPath) {
+			emmetExtensionsPath = emmetExtensionsPath.trim();
 		}
-	} else if (!isAbsolutePath(emmetExtensionsPath)) {
-		if (workspaceFolderPath) {
-			emmetExtensionsPathUri = joinPath(workspaceFolderPath, emmetExtensionsPath);
+		if (!emmetExtensionsPath) {
+			resetSettingsFromFile();
+			return Promise.resolve();
 		}
-	} else {
-		emmetExtensionsPathUri = URI.file(emmetExtensionsPath);
-	}
 
-	try {
-		// the fs.stat call itself could throw, so we wrap this part up into a try-catch
-		if (!emmetExtensionsPathUri || (await fs.stat(emmetExtensionsPathUri)).type !== FileType.Directory) {
-			throw new Error();
+		if (emmetExtensionsPath[0] === '~') {
+			if (homeDir) {
+				emmetExtensionsPathUri = joinPath(homeDir, emmetExtensionsPath.substr(1));
+			}
+		} else if (!isAbsolutePath(emmetExtensionsPath)) {
+			if (workspaceFolderPath) {
+				emmetExtensionsPathUri = joinPath(workspaceFolderPath, emmetExtensionsPath);
+			}
+		} else {
+			emmetExtensionsPathUri = URI.file(emmetExtensionsPath);
 		}
-	} catch (e) {
+
+		try {
+			// the fs.stat call itself could throw, so we wrap this part up into a try-catch
+			if (!emmetExtensionsPathUri || (await fs.stat(emmetExtensionsPathUri)).type !== FileType.Directory) {
+				throw new Error();
+			}
+		} catch (e) {
+			continue;
+		}
+		findValidPath = true;
```

Comment:
I'd name this variable `hasValidPath` since it's a boolean variable. `findValidPath` sounds more like a function name.